mms/model_loader.py

```diff
@@ -45,28 +48,45 @@ def download(url, path=None, overwrite=False):
     str
         The file path of the downloaded file.
     """
-    if path is None:
-        fname = url.split('/')[-1]
-    elif os.path.isdir(path):
-        fname = os.path.join(path, url.split('/')[-1])
-    else:
-        fname = path
-
-    if overwrite or not os.path.exists(fname):
-        dirname = os.path.dirname(os.path.abspath(os.path.expanduser(fname)))
-        if not os.path.exists(dirname):
-            os.makedirs(dirname)
-
-        print('Downloading %s from %s...' % (fname, url))
-        r = requests.get(url, stream=True)
-        if r.status_code != 200:
-            raise RuntimeError("Failed downloading url %s" % url)
-        with open("%s.temp" % (fname), 'wb') as f:
-            for chunk in r.iter_content(chunk_size=1024):
-                if chunk:  # filter out keep-alive new chunks
-                    f.write(chunk)
-        os.rename("%s.temp" % (fname), fname)
-    return fname
+    model_file = model_location
+    if model_location.lower().startswith(URL_PREFIX):
+        if path is None:
+            model_file = model_location.split('/')[-1]
+        elif os.path.isdir(path):
+            model_file = os.path.join(path, model_location.split('/')[-1])
+        else:
+            model_file = path
+
+        if overwrite or not os.path.exists(model_file):
```

Comment:
each worker process will download and extract model to same location, duplicating work.

the lock will serialize access to this function, so results will be correct, but startup time will be multiplied by # of workers.

Reply:
Yes, model will be extracted and download multiple times. But it should be according to the overwrite parameter. If we change the default behavior to always check existence, we are breaking the flow.

Reply:
ok we will fix that separately